### ISSUE TITLE
Memory issue validation

### ISSUE THREAD

**Issue by DiegoOrtego** [ZijunDeng/pytorch-semantic-segmentation#13]
Hi @ZijunDeng , I am reusing your PSPNet implementation for my purposes and I noticed a problematic situation when validating with a large set of images. You append in a list all predictions and gts in order to compute the performance metrics. So if the dataset is large enough, the memory is easily saturated.

**Comment by ZijunDeng** [comment]
Yes, that's a problem. The evaluation of mIOU has to be done at once on the whole val set. My computer has 32GB memory, which is enough.
